rpc/client/dcrd/calls.go

```diff
@@ -277,6 +284,54 @@ func (r *RPC) CFilters(ctx context.Context, blockHashes []*chainhash.Hash) ([]*g
 	return filters, nil
 }
 
+// CFilterV2 returns the version 2 committed filter and the data required for
+// verifying the inclusion proof of the cfilter for a block.
+func (r *RPC) CFilterV2(ctx context.Context, blockHash *chainhash.Hash) (*gcs2.FilterV2, uint32, []chainhash.Hash, error) {
+	const opf = "dcrd.CFilterV2(%v)"
+
+	var res cfilterV2Reply
+	err := r.Call(ctx, "getcfilterv2", &res, blockHash.String())
+	if err != nil {
+		op := errors.Opf(opf, blockHash)
+		return nil, 0, nil, errors.E(op, err)
+	}
+
+	return res.Filter.Filter, res.ProofIndex, res.proofHashes(), nil
+}
+
+// CFiltersV2 returns the version 2 committed filters for blocks.
+//
+// FIXME: Returning a []func() (...) is an ugly hack to be able to avoid the
+// circular dependency between the wallet pkg and this pkg and have it fulfill
+// an interface defined there.
```

Comment:
I'm not seeing what the issue is here. Can you elaborate about this?

Reply:
The current dependency chain for packages is `spv` -> `wallet` -> `rpc` (actually `rpc/client/dcrd`).

So I can't make `rpc` and `spv` both return some struct defined in `wallet`. I could make `spv` depend on `rpc` but that feels wrong. I could also introduce a third package but seems way too much just for this one struct.

I think that in the long term, removing the `wallet` -> `rpc` coupling would be best (and have the `Network` interface be generic enough for all needs) but obviously not something we want in this PR.

Reply:
Ah, i see now, and you are using a func to return the fields that would otherwise be part of this struct.

Maybe a type alias would work in wallet:

```
type FilterProof = struct {
    // ...
}
```

then I believe other packages which wallet imports would be able to satisfy this interface.

This feels like a hack however.

Reply:
Less hacky than returning a slice of fuctions IMO, so I switched to that.

Reply:
comment needs to be updated now since this was changed